packages/datetime/src/dateRangeInput.tsx

```diff
@@ -362,6 +331,30 @@ export class DateRangeInput extends AbstractComponent<IDateRangeInputProps, IDat
         this.setState(nextState);
     }
 
+    private renderInputGroup = (boundary: DateRangeBoundary) => {
+        const inputProps = boundary === DateRangeBoundary.START ? this.props.startInputProps : this.props.endInputProps;
+        const htmlProps = this.getHtmlPropsFromInputProps(inputProps);
+
+        const handleInputEvent = (e: React.SyntheticEvent<HTMLInputElement>) => this.handleInputEvent(e, boundary);
```

Comment:
this isn't any better than inlining the function in JSX, it's still a new instance every time this function is called. any way we can store the two instances instead of re-creating every render?